### PR TITLE
feat(ci)_: add Jenkinsfil for RPC integration tests

### LINKED ISSUE #15148: Extend docker-compose.anvil.yml with deployment of remaining status-im contracts
we should extend [docker-compose.anvil.yml ](https://github.com/status-im/status-go/blob/develop/integration-tests/docker-compose.anvil.yml)to deploy following contracts to anvil:
* https://github.com/status-im/community-dapp
* https://github.com/status-im/sticker-market
* https://github.com/status-im/ens-usernames
* https://github.com/status-im/status-go/blob/ea5c444dbee158a96a8db0d9373ec15e76629e80/contracts/balancechecker/BalanceChecker.sol

Other contracts inside status-go:

- [balancechecker](https://github.com/status-im/status-go/tree/develop/contracts/balancechecker)
- [directory](https://github.com/status-im/status-go/tree/develop/contracts/directory)
- [ethscan](https://github.com/status-im/status-go/tree/develop/contracts/ethscan)
- [ierc20](https://github.com/status-im/status-go/tree/develop/contracts/ethscan)
- [registrar](https://github.com/status-im/status-go/tree/develop/contracts/registrar)
- [resolver](https://github.com/status-im/status-go/tree/develop/contracts/resolver)
- [snt](https://github.com/status-im/status-go/tree/develop/contracts/snt)
- [stickers](https://github.com/status-im/status-go/tree/develop/contracts/stickers)

### PR DESCRIPTION
**Pull request by jakubgs**
This PR will enable status-go RPC integration tests to run against every commit using following workflow:

- spin up anvil
- deploy [communities-contracts ](https://github.com/status-im/communities-contracts)
- deploy[ status-network-token-v2](https://github.com/status-im/status-network-token-v2) 
- build and start status-go as daemon with anvil as RPC endpoint
- run RPC wallet tests

Running time for RPC tests including build time is ~3 minutes, we are committing to keep it reasonable which is less then 5 minutes
More info regarding RPC tests https://github.com/status-im/status-go/pull/5302

Please note, more contracts will be added in future as part of https://github.com/status-im/status-desktop/issues/15148
Please note, more RPC tests including not only wallet will be added in future e.g. https://github.com/status-im/status-go/pull/5413

### CODE REVIEW

File: integration-tests/config.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "HTTPEnabled": true,
     "HTTPHost": "0.0.0.0",
     "HTTPPort": 3333,
+    "HTTPVirtualHosts": ["*", "status-go"],
```
Comment (igor-sirotin) [inline integration-tests/config.json:10]: `"*"` sounds a bit excessive, no?

File: api/default_test.go [hunk 2]
```diff
@@ -11,14 +11,15 @@ import (
...
-		HTTPPort:        8545,
-		WSEnabled:       false,
-		WSHost:          "127.0.0.1",
-		WSPort:          8586,
+		APIModules:       "test, eth, wakuv2",
+		ConnectorConfig:  params.ConnectorConfig{Enabled: true},
+		HTTPEnabled:      true,
+		HTTPHost:         "0.0.0.0",
+		HTTPPort:         8545,
+		HTTPVirtualHosts: []string{"status-go"},
```
Comment (igor-sirotin) [inline api/default_test.go:19]: Is `"status-go"` the default value? Not an empty list?

**Review by igor-sirotin — APPROVED** [review]
It would be super-cool to integrate these tests to coverage reports.
https://go.dev/blog/integration-test-coverage

But:
1. It's a very early stage, let's gather some tests first
2. We'd need to merge results from `tests` and `tests-rpc` runs, which are separate now.

File: api/default_test.go (hunk 2 above)
Comment (jakubgs) [reply api/default_test.go:19]: Honestly, we should just use `"*"` for tests and not think about it.

File: integration-tests/config.json (hunk 1 above)
Comment (jakubgs) [reply integration-tests/config.json:10]: I don't see an issue for tests.

**Comment by IvanBelyakoff** [comment]
We have discussed it with @antdanchenko in DM, that we should not override `HTTPVirtualHosts` value of node config from Login request, as AFAIK that will allow RPC requests from any hosts afterwards, even if this login request was a fake one. So we might need to inject a testing "overrideAPIConfig" function for tests.

**Review by IvanBelyakoff — APPROVED** [review]

**Comment by igor-sirotin** [comment]
jsut in case, it would be nice to see the results on `Tests` jenkins page:
https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests-rpc/detail/PR-5428/19/tests

image

**Comment by IvanBelyakoff** [comment]
> jsut in case, it would be nice to see the results on `Tests` jenkins page: https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests-rpc/detail/PR-5428/19/tests
> 
> image

I did not notice a test that Anton made and broke it, fixed just now. Should be good.

**Comment by antdanchenko** [comment]
> jsut in case, it would be nice to see the results on `Tests` jenkins page: https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests-rpc/detail/PR-5428/19/tests
> 
Screenshot 2024-07-04 at 12 02 01

added results

**Pull request merged**